src/test/java/io/spokestack/spokestack/tts/TTSCallbackTest.java

```diff
@@ -71,16 +74,54 @@ public void testQueuing() throws IOException {
 
         callback.clear();
 
-        // out-of-order responses
-        callback.enqueue(requestId, 500L);
-        callback.enqueue(request2Id, 500L);
-        callback.onResponse(call, response2);
-        callback.onResponse(call, response);
-        assertEquals(2, callback.responses.size());
-        audioResponse = callback.responses.get(0);
-        assertEquals(requestId, audioResponse.getMetadata().get("id"));
-        audioResponse = callback.responses.get(1);
-        assertEquals(request2Id, audioResponse.getMetadata().get("id"));
+        // out-of-order responses from different threads
+        List<String> requestIds = new ArrayList<>();
+        List<Response> thread1Responses = new ArrayList<>();
+        List<Response> thread2Responses = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            requestId = "request" + i;
+            requestIds.add(requestId);
+            callback.enqueue(requestId, 500L);
+            request = createRequest(requestId);
+            response = TTSTestUtils.createHttpResponse(request, responseBody);
+            if (i < 5) {
+                thread1Responses.add(response);
+            } else {
+                thread2Responses.add(response);
+            }
+        }
+
+        // chosen by random dice roll
+        Random random = new Random(4);
+        Collections.shuffle(thread1Responses, random);
+        Collections.shuffle(thread2Responses, random);
+
+        CountDownLatch latch = new CountDownLatch(2);
+        responseThread(call, thread1Responses, callback, latch).start();
+        responseThread(call, thread2Responses, callback, latch).start();
```

Comment:
Sadly, I can't get this new version of the test to fail even with the old plain `List` version of the queue. Even tried adding a sleep after each call to `onResponse` to try to ensure that at least some of the calls would be interleaved. Can you think of another organization here that might cause concurrency issues in the callback?

Reply:
it's going to be a rare race condition, so it will be hard to reproduce in a test